app/brave_main_delegate.cc

```diff
@@ -212,5 +205,23 @@ std::optional<int> BraveMainDelegate::PostEarlyInitialization(
         switches::kComponentUpdater,
         (current_value + "url-source=" + update_url).c_str());
   }
+
+  // For Self-host sync service URL
+  if (command_line->HasSwitch(syncer::kSyncServiceURL)) {
+    GURL sync_service_url =
+        GURL(command_line->GetSwitchValueASCII(syncer::kSyncServiceURL));
+    // We validate the URL to ensure it meets security requirements:
+    // 1. The URL must be valid
+    // 2. The URL must use HTTPS (or be otherwise potentially trustworthy like
+    // localhost) If the URL doesn't meet these requirements, we remove the
+    // switch and use the default sync URL
+    if (!sync_service_url.is_valid() ||
+        !sync_service_url.SchemeIsHTTPOrHTTPS() ||
+        !network::IsOriginPotentiallyTrustworthy(
+            url::Origin::Create(sync_service_url))) {
+      command_line->RemoveSwitch(syncer::kSyncServiceURL);
```

Comment:
I'm not sure what is the practice, but if we are removing the switch due to security concerns, shouldn't we maybe `LOG(WARNING)`, or would that be too spammy because of services?